cc2dataset/main.py

```diff
@@ -12,13 +12,131 @@
 from pyspark import SparkContext
 from pyspark.sql.functions import rand
 from pyspark.sql import SparkSession
+from bs4 import BeautifulSoup
 import random
 import math
 import time
 from .spark_session_builder import build_spark_session
 from io import BytesIO
 from urllib.parse import urljoin
+from yt_dlp.extractor import gen_extractor_classes, GenericIE
+from urllib.parse import urlparse
+import traceback
 
+import re
+def is_youtube_video(url):
+  if re.match('^https?://(www.)?youtube.com/watch\?v=.+$', url):
+    return True
+  if re.match('^https?://(www.)?youtube.com/v/.+$', url):
+    return True
+  if re.match('^https?://(www.)?youtube.com/embed/.+$', url):
+    return True
+  if re.match('^https?://(www.)?youtu.be/.+$', url):
+    return True
+
+  return False
+
+
+def is_bilibili_video(url):
+    if re.match("https?://(?:www\.)?bilibili\.com/(?:video/|festival/\w+\?(?:[^#]*&)?bvid=)[aAbB][vV](?P<id>[^/?#&]+)",url):
+        return True
+
+    return False
+
+def valid_video_platform_link_(link):
+    return is_bilibili_video(link.get("url", ""))
+
+import yt_dlp
+import unicodedata
+
+generic_extractors = [yt_dlp.extractor.generic.GenericIE, yt_dlp.extractor.lazy_extractors.GenericIE]
+porn_patterns = ["porn", "adult", "xxx", "xvideos", "xhamster", "redtube", "xtube", "xstream", "xfileshare", "sex"]
```

Comment:
add "tnaflix" in the list of porn patterns

Reply:
we need also to exclude http://xnxx.com, i.e. add "xnxx" in the list of porn patterns